transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -842,10 +843,8 @@ public ChannelFuture writeAndFlush(Object msg) {
     }
 
     private static void notifyOutboundHandlerException(Throwable cause, ChannelPromise promise) {
-        if (!promise.tryFailure(cause) && !(promise instanceof VoidChannelPromise)) {
-            if (logger.isWarnEnabled()) {
-                logger.warn("Failed to fail the promise because it's done already: {}", promise, cause);
-            }
+        if (!(promise instanceof VoidChannelPromise)) {
```

Comment:
This will not work as `PromiseNotificationUtil ` is in the common module. That said I think I could add a `VoidPromise` interface in the common package, mark it internal and check for it. WDYT ?